### PR TITLE
initial attempt to support archiving segements in Vault

### PR DESCRIPTION
**Pull request by hghaf099**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,8 +10,8 @@ require (
 	github.com/benmathews/hdrhistogram-writer v0.0.0-20210120211942-3cb1c7c33f95
 	github.com/coreos/etcd v3.3.27+incompatible
 	github.com/google/gofuzz v1.2.0
-	github.com/hashicorp/go-hclog v0.14.1
-	github.com/hashicorp/raft v1.4.0
+	github.com/hashicorp/go-hclog v1.5.0
+	github.com/hashicorp/raft v1.5.0
```
Comment (ncabatoff) [inline go.mod:14]: Is this needed?

File: segment/writer.go [hunk 2]
```diff
@@ -413,6 +413,7 @@ func (w *Writer) appendIndex() error {
 	// Record the file offset where the index starts (the actual index data so
 	// after the frame header).
 	w.writer.indexStart = uint64(w.writer.writeOffset) + uint64(startOff+frameHeaderLen)
+	fmt.Println("Hamid - in appendIndex - index start of the write is: ", w.writer.indexStart)
```
Comment (hghaf099) [inline segment/writer.go:416]: I will remove these!

File: wal_test.go [hunk 3]
```diff
@@ -965,3 +965,31 @@ func TestClose(t *testing.T) {
...
+	require.NoError(t, err)
+
+	t.Log(ts.debugDump())
+
+	sealed, err := w.GetSealedLogFiles(200)
+	if err != nil {
+		t.Fatalf("failed to get sealed logs: %v", err)
+	}
+
+	if len(sealed) != 3 {
```
Comment (ncabatoff) [inline wal_test.go:992]: Why 3 rather than 4?  I gather it has to do with the fromIndex=200, but if you could add a hint as a comment it would be nice.

Comment (hghaf099) [reply wal_test.go:992]: Yeah, exactly, fromIndex points to a baseIndex of a segment which is included in the result, and the sum is only three segments. The last one is the tail which is not sealed yet.

Comment (hghaf099) [reply wal_test.go:992]: I will add a comment!

**Comment by hghaf099** [comment]
Closing, as this was done for a hackweek project.

**Pull request closed without merging**